packages/web3-eth-abi/src/types.ts

```diff
@@ -0,0 +1,57 @@
+export interface JsonAbiParameter {
+	readonly name: string;
+	readonly type: string;
+	readonly baseType?: string;
+	readonly indexed?: boolean;
+	readonly components?: Array<JsonAbiParameter>;
+	readonly arrayLength?: number;
+	readonly arrayChildren?: Array<JsonAbiParameter>;
+}
+
+export interface JsonAbiStruct {
+	[key: string]: unknown;
+	name?: string;
+	type: string;
+}
+
+export interface JsonAbiCoderStruct extends JsonAbiStruct {
+	[key: string]: unknown;
+	components?: Array<JsonAbiStruct>;
+}
+
+type FragmentTypes = 'constructor' | 'event' | 'function';
+
+export interface JsonAbiBaseFragment {
+	[key: string]: unknown;
+	name?: string;
+	type: FragmentTypes;
+	inputs?: Array<JsonAbiParameter>;
+}
+
+export interface JsonAbiConstructorFragment extends JsonAbiBaseFragment {
```

Comment:
good to add source in doc comments for future developers https://docs.soliditylang.org/en/latest/abi-spec.html#json

Reply:
Added to union type.